tee-worker/ts-tests/common/utils.ts

```diff
@@ -397,3 +304,145 @@ export async function checkFailReason(
+                    return item;
+                }
+            }
+        }
+    }
+    return null;
+}
+
+export function buildStorageKey(
+    metadata: Metadata,
+    prefix: string,
+    method: string,
+    keyTypeId?: SiLookupTypeId,
+    hashers?: Array<StorageHasherV14>,
+    input?: Array<unknown>
+): Uint8Array {
+    let storageKey = u8aConcat(xxhashAsU8a(prefix, 128), xxhashAsU8a(method, 128));
+    if (keyTypeId && hashers && input) {
+        let keyTypeIds = hashers.length === 1 ? [keyTypeId] : metadata.registry.lookup.getSiType(keyTypeId).def.asTuple;
+        for (let i = 0; i < keyTypeIds.length; i++) {
+            const theKeyTypeId = keyTypeIds[i];
+            const theHasher = hashers[i].toString();
+            const theKeyItem = input[i];
+            // get the scale encoded input data by encoding the input
+            const theKeyType = metadata.registry.createLookupType(theKeyTypeId);
+            const theKeyItemEncoded = metadata.registry.createType(theKeyType, theKeyItem).toU8a();
+            // apply hasher
+            let theKeyItemAppliedHasher;
+            if (theHasher == 'Blake2_128Concat') {
+                theKeyItemAppliedHasher = blake2128Concat(theKeyItemEncoded);
+            } else if (theHasher == 'Twox64Concat') {
+                theKeyItemAppliedHasher = twox64Concat(theKeyItemEncoded);
+            } else if (theHasher == 'Identity') {
+                theKeyItemAppliedHasher = identity(theKeyItemEncoded);
+            } else {
+                throw new Error(`The hasher ${theHasher} is not support.`);
+            }
+            storageKey = u8aConcat(storageKey, theKeyItemAppliedHasher);
+        }
+    }
+    return storageKey;
+}
+export async function buildStorageData(
+    metadata: Metadata,
+    prefix: string,
+    method: string,
+    ...input: Array<unknown>
+): Promise<string | null> {
+    const storageEntry = buildStorageEntry(metadata, prefix, method);
+    if (!storageEntry) {
+        throw new Error('Can not find the storage entry from metadata');
+    }
+    let storageKey, valueType;
+
+    if (storageEntry.type.isPlain) {
+        storageKey = buildStorageKey(metadata, prefix, method);
+        valueType = metadata.registry.createLookupType(storageEntry.type.asPlain);
+    } else if (storageEntry.type.isMap) {
+        const { hashers, key, value } = storageEntry.type.asMap;
+        if (input.length != hashers.length) {
+            throw new Error('The `input` param is not correct');
+        }
+        storageKey = buildStorageKey(metadata, prefix, method, key, hashers, input);
+        valueType = metadata.registry.createLookupType(value);
+    } else {
+        throw new Error('Only support plain and map type');
+    }
+    console.debug(`storage key: ${u8aToHex(storageKey)}`);
+    return u8aToHex(storageKey);
+}
+
+export async function checkUserShieldingKeys(
+    context: IntegrationTestContext,
+    pallet: string,
+    method: string,
```

Comment:
We can hardcode the `pallet` and `method` names - they are unlikely to change, only the `address` can verify (i.e. as per which address I want to check).

It will reduce a lot of duplicate code. Same for challengeCode and idGraphs.

This function is specific to our code anyway.